src/PIL/ImageOps.py

```diff
@@ -623,6 +623,100 @@ def grayscale(image: Image.Image) -> Image.Image:
     return image.convert("L")
 
 
+def sepia(image: Image.Image) -> Image.Image:
+    """
+    Apply a sepia tone effect to an image.
+
+    :param image: The image to modify.
+    :return: An image.
+
+    """
+    if image.mode != "RGB":
+        image = image.convert("RGB")
+
+    out = Image.new("RGB", image.size)
+
+    for x in range(image.width):
+        for y in range(image.height):
+            value = image.getpixel((x, y))
+            assert isinstance(value, tuple)
+            r, g, b = value
+
+            tr = 0.393 * r + 0.769 * g + 0.189 * b
+            tg = 0.349 * r + 0.686 * g + 0.168 * b
+            tb = 0.272 * r + 0.534 * g + 0.131 * b
+
+            out.putpixel((x, y), tuple(min(255, int(c)) for c in (tr, tg, tb)))
+
+    return out
+
+
+def sobel(image: Image.Image) -> Image.Image:
+    """
+    Applies a Sobel edge-detection filter to the given image.
+
+    This function computes the Sobel gradient magnitude using the
+    horizontal (Gx) and vertical (Gy) Sobel kernels.
+
+    :param image: the image to apply the filter
+    :return: An image.
+    """
+    if image.mode != "L":
+        image = image.convert("L")
+
+    Kx = [[-1, 0, 1], [-2, 0, 2], [-1, 0, 1]]
+    Ky = [[1, 2, 1], [0, 0, 0], [-1, -2, -1]]
+
+    out = Image.new("L", image.size)
+
+    for y in range(1, image.height - 1):
+        for x in range(1, image.width - 1):
```

Comment:
```suggestion
    for x in range(1, image.width - 1):
        for y in range(1, image.height - 1):
```
Nitpick: I think it would be nice if the order of the loops matched the order in `sepia()`